vespa-feed-client/src/main/java/ai/vespa/feed/client/impl/ApacheCluster.java

```diff
@@ -160,48 +163,48 @@ private Endpoint(CloseableHttpAsyncClient client, URI url) {
 
     }
 
-    @SuppressWarnings("deprecation")
     private static CloseableHttpAsyncClient createHttpClient(FeedClientBuilderImpl builder) throws IOException {
         SSLContext sslContext = builder.constructSslContext();
         String[] allowedCiphers = excludeH2Blacklisted(excludeWeak(sslContext.getSupportedSSLParameters().getCipherSuites()));
         if (allowedCiphers.length == 0)
             throw new IllegalStateException("No adequate SSL cipher suites supported by the JVM");
 
         ClientTlsStrategyBuilder tlsStrategyBuilder = ClientTlsStrategyBuilder.create()
-                .setTlsDetailsFactory(TlsDetailsFactory::create)
                 .setCiphers(allowedCiphers)
                 .setSslContext(sslContext);
         if (builder.hostnameVerifier != null)
             tlsStrategyBuilder.setHostnameVerifier(builder.hostnameVerifier);
 
-        return HttpAsyncClients.createHttp2Minimal(H2Config.custom()
-                                                           .setMaxConcurrentStreams(builder.maxStreamsPerConnection)
-                                                           .setCompressionEnabled(true)
-                                                           .setPushEnabled(false)
-                                                           .setInitialWindowSize(Integer.MAX_VALUE)
-                                                           .build(),
-                                                   IOReactorConfig.custom()
-                                                                  .setIoThreadCount(2)
-                                                                  .setTcpNoDelay(true)
-                                                                  .setSoTimeout(Timeout.ofSeconds(10))
-                                                                  .build(),
-                                                   tlsStrategyBuilder.build());
+        // Socket timeout must be longer than the longest feasible response timeout
+        Timeout socketTimeout = Timeout.ofMinutes(15);
```

Comment:
Why?

Reply:
The request/stream will be failed out on socket timeout